src/components/SingleName/ResolverAndRecords/RecordsItem.js

```diff
@@ -226,202 +227,191 @@ const RecordItemEditable = ({
-              <>
-                <IpfsLogin startAuthorizing={startAuthorizing} />
-                <SaveCancel
-                  warningMessage={getOldContentWarning(
-                    type,
-                    domain.contentType
-                  )}
-                  mutation={e => {
-                    e.preventDefault()
-                    const variables = {
-                      name: domain.name,
-                      [variableName ? variableName : 'recordValue']: newValue
-                    }
-                    mutation({
-                      variables
-                    })
-                  }}
+                  dataType={type}
+                  contentType={domain.contentType}
+                  updateValue={updateValue}
                   isValid={isValid}
-                  stopEditing={stopUploading}
+                  isInvalid={isInvalid}
                 />
-              </>
-            ) : (
-              ''
-            )}
-          </RecordsItem>
+              )}
+            </EditRecord>
+            <SaveCancel
+              warningMessage={getOldContentWarning(type, domain.contentType)}
+              mutation={e => {
+                e.preventDefault()
+                const variables = {
+                  name: domain.name,
+                  [variableName ? variableName : 'recordValue']: newValue
+                }
+                executeMutation({
+                  variables
+                })
+              }}
+              isValid={isValid}
+              stopEditing={stopEditing}
+            />
+          </>
+        ) : uploading && authorized ? (
+          <>
+            <EditRecord>
+              <Upload updateValue={updateValue} newValue={newValue} />
+              {newValue !== '' && (
+                <NewRecordsContainer>
+                  <RecordsKey>New IPFS Hash:</RecordsKey>
+                  <ContentHashLink
+                    value={newValue}
+                    contentType={domain.contentType}
+                    domain={domain}
+                  />
+                </NewRecordsContainer>
+              )}
+            </EditRecord>
+            <SaveCancelSwitch
+              warningMessage={getOldContentWarning(type, domain.contentType)}
+              mutation={e => {
+                e.preventDefault()
+                const variables = {
+                  name: domain.name,
+                  [variableName ? variableName : 'recordValue']: newValue
+                }
+                executeMutation({
+                  variables
+                })
+              }}
+              isValid={isValid}
+              newValue={newValue}
+              startUploading={startUploading}
+              stopUploading={stopUploading}
+              stopAuthorizing={stopAuthorizing}
+            />
+          </>
+        ) : uploading && !authorized ? (
+          <>
+            <IpfsLogin startAuthorizing={startAuthorizing} />
```

Comment:
We can probably remove this as I guess it's not used? But maybe leave it for a separate PR/issue